shared/chat/conversation/fwd-msg/team-picker.tsx

```diff
@@ -0,0 +1,181 @@
+import * as Chat2Gen from '../../../actions/chat2-gen'
+import * as Types from '../../../constants/types/chat2'
+import * as Constants from '../../../constants/chat2'
+import {Avatars, TeamAvatar} from '../../avatars'
+import debounce from 'lodash/debounce'
+import logger from '../../../logger'
+
+type Props = Container.RouteProps<{srcConvID: Types.ConversationIDKey; ordinal: Types.Ordinal}>
+
+const TeamPicker = (props: Props) => {
+  const srcConvID = Container.getRouteProps(props, 'srcConvID', '')
+  const ordinal = Container.getRouteProps(props, 'ordinal', 0)
+  const message = Container.useSelector(state => Constants.getMessage(state, srcConvID, ordinal))
+  const [term, setTerm] = React.useState('')
+  const [results, setResults] = React.useState<Array<RPCChatTypes.AddBotConvSearchHit>>([])
+  const [waiting, setWaiting] = React.useState(false)
+  const [error, setError] = React.useState('')
+  const fwdMsg = Container.useRPC(RPCChatTypes.localForwardMessageNonblockRpcPromise)
+  const submit = Container.useRPC(RPCChatTypes.localAddBotConvSearchRpcPromise)
+  const dispatch = Container.useDispatch()
+  const doSearch = React.useCallback(() => {
+    setWaiting(true)
+    submit(
+      [{term}],
+      result => {
+        setWaiting(false)
+        setResults(result ?? [])
+      },
+      error => {
+        setWaiting(false)
+        setError('Something went wrong, please try again.')
+        logger.info('TeamPicker: error loading search results: ' + error.message)
+      }
+    )
+  }, [term, submit])
+
+  const onClose = () => {
+    dispatch(RouteTreeGen.createClearModals())
+  }
+  const onSelect = (dstConvID: RPCChatTypes.ConversationID) => {
+    if (!message) {
+      setError('Something went wrong, please try again.')
+      return
+    }
+    if (message.type == 'attachment') {
+      dispatch(
+        Chat2Gen.createAttachmentDownload({
+          message,
+        })
+      )
+    }
+    fwdMsg(
+      [
+        {
+          srcConvID: Types.keyToConversationID(srcConvID),
+          msgID: message.id,
+          dstConvID,
+          identifyBehavior: RPCTypes.TLFIdentifyBehavior.chatGui,
+        },
+      ],
+      () => {
+        setWaiting(false)
+      },
+      error => {
+        setWaiting(false)
+        setError('Something went wrong, please try again.')
+        logger.info('TeamPicker: error loading search results: ' + error.message)
+      }
+    )
+    dispatch(RouteTreeGen.createClearModals())
+  }
+  React.useEffect(() => {
+    doSearch()
+  }, [doSearch])
+
+  const renderResult = (index: number, item: RPCChatTypes.AddBotConvSearchHit) => {
+    return (
+      <Kb.ClickableBox key={index} onClick={() => onSelect(item.convID)}>
+        <Kb.Box2 direction="horizontal" fullWidth={true} gap="tiny" style={styles.results}>
+          {item.isTeam ? (
+            <TeamAvatar isHovered={false} isMuted={false} isSelected={false} teamname={item.name} />
```

Comment:
a little unclear what this does with a big team, might be good to append '#general' if we're only supporting forwarding to the default convo for teams

Reply:
oops this should support channels too